services/QuillLMS/spec/services/analytics/cache_segment_school_data_spec.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CacheSegmentSchoolData do
+  let(:school) { create(:school) }
+  let(:teacher1) { create(:teacher, last_sign_in: Time.zone.today - 366.days) }
+  let(:teacher2) { create(:teacher, last_sign_in: Time.zone.today) }
+  let!(:schools_users1) { create(:schools_users, school: school, user: teacher1 )}
+  let!(:schools_users2) { create(:schools_users, school: school, user: teacher2 )}
+  let(:classrooms_teachers1) { create(:classrooms_teacher, user: teacher1) }
+  let(:classrooms_teachers2) { create(:classrooms_teacher, user: teacher2) }
+  let(:students_classrooms1) { create_list(:students_classrooms, 10, classroom: classrooms_teachers1.classroom)}
+  let(:students_classrooms2) { create_list(:students_classrooms, 10, classroom: classrooms_teachers2.classroom)}
+  let!(:activity_sessions1) { create_list(:activity_session, 20, user: students_classrooms1.map(&:student).sample, completed_at: Time.zone.today - 367.days ) }
+  let!(:activity_sessions2) { create_list(:activity_session, 20, user: students_classrooms2.map(&:student).sample, completed_at: Time.zone.today ) }
+
+  before do
+    students_classrooms1.each do |sc|
+      sc.student.update(last_sign_in: Time.zone.today - 366.days)
+    end
+    students_classrooms2.each do |sc|
+      sc.student.update(last_sign_in: Time.zone.today)
+    end
+  end
+
+  subject { described_class.new(school) }
+
+  # rubocop:disable RSpec/SubjectStub
```

Comment:
I think this linter rule is misfiring - we were getting the `Rspec/SubjectStub` error on every `expect(...)to receive` line in this test file, even though none of them are stubbing. Please let me know if I'm missing something here.